src/js/grapl-cdk/lib/grapl-cdk-stack.ts.orig

```diff
@@ -0,0 +1,539 @@
+                    "BUCKET_PREFIX": prefix
+                },
+                timeout: cdk.Duration.seconds(25),
+                memorySize: 256,
+                description: grapl_version,
+            }
+        );
+        this.event_handler.currentVersion.addAlias('live');
+
+        if (this.event_handler.role) {
+            jwt_secret.grantRead(this.event_handler.role);
+            user_auth_table.allowReadFromRole(this.event_handler.role);
+
+            model_plugin_bucket.grantReadWrite(this.event_handler.role);
+            model_plugin_bucket.grantDelete(this.event_handler.role);
+        }
+
+        this.integration = new apigateway.LambdaRestApi(
+            this,
+            this.integrationName,
+            {
+                handler: this.event_handler,
+            },
+        );
+
+        this.integration.addUsagePlan('integrationApiUsagePlan', {
+            quota: {
+                limit: 1000,
+                period: apigateway.Period.DAY,
+            },
+            throttle: {  // per minute
+                rateLimit: 50,
+                burstLimit: 50,
+            }
+        });
+    }
+}
+
+
+export interface GraplEnvironementProps {
+    prefix: string,
+    jwt_secret: secretsmanager.Secret,
+    vpc: ec2.IVpc,
+    master_graph: DGraphEcs,
+    user_auth_table: UserAuthDb,
+}
+
+export class GraplCdkStack extends cdk.Stack {
+    grapl_env: GraplEnvironementProps;
+
+    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+        super(scope, id, props);
+
+        const prefix = process.env.BUCKET_PREFIX || "my";
+
+        const mgZeroCount = Number(process.env.MG_ZEROS_COUNT) || 1;
+        const mgAlphaCount = Number(process.env.MG_ALPHAS_COUNT) || 1;
+
+        const grapl_vpc = new ec2.Vpc(this, 'VPC', {
+            natGateways: 1,
+            enableDnsHostnames: true,
+            enableDnsSupport: true,
+        });
+
+
+        const jwtSecret = new secretsmanager.Secret(this, 'EdgeJwtSecret', {
+            description: 'The JWT secret that Grapl uses to authenticate its API',
+            secretName: 'EdgeJwtSecret',
+        });
+
+        const user_auth_table = new UserAuthDb(this, 'UserAuthTable');
+
+        const analyzers_bucket = new s3.Bucket(this, prefix + '-analyzers-bucket', {
+            bucketName: prefix + '-analyzers-bucket',
+            removalPolicy: cdk.RemovalPolicy.DESTROY,
+            encryption: BucketEncryption.KMS_MANAGED,
+            blockPublicAccess: BlockPublicAccess.BLOCK_ALL
+        });
+
+        const engagements_created_topic =
+            new sns.Topic(this, id + '-engagements-created-topic', {
+                topicName: 'engagements-created-topic'
+            });
+
+        const master_graph = new DGraphEcs(
+            this,
+            'master-graph',
+            grapl_vpc,
+            mgZeroCount,
+            mgAlphaCount,
+        );
+
+<<<<<<< HEAD
```

Comment:
Looks like a merge conflict wasn't fully resolved in this file

Reply:
gah - this file shouldn't even be there. good catch.